miner/worker.go

```diff
@@ -594,11 +594,12 @@ func (w *worker) mainLoop() {
 		}
 	}()
 
+	bor, _ := w.engine.(*bor.Bor)
```

Comment:
It might be worth checking the type assertion here. E.g.

```golang
bor, isBor := w.engine.(*bor.Bor)
devFakeAuthor := isBor && bor != nil && bor.DevFakeAuthor
// ....
if w.eth.PeerCount() > 0 || devFakeAuthor {
      // ...
}

```

Reply:
This did occur to me but I didn't pay much heed because we don't expect any consensus engine other than bor running.